### ISSUE TITLE
[Development] Form 526 - FE update rated disabilities

### ISSUE THREAD

**Issue by annaswims** [department-of-veterans-affairs/va.gov-team#13128]
## Issue Description
When a veteran creates a form526 we [prefill](https://github.com/department-of-veterans-affairs/vets-api/blob/c559b92a521a6614fc087316a09dd3b75235fee4/app/models/form_profiles/va526ez.rb#L72) rated_disabilities_information when a veteran starts a form.  However, it's possible that the veterans rated disability info will be updated in the upstream system before the veteran submits the form, especially if they start the form and come back to it later. 

~We already have an endpoint  ` /v0/disability_compensation_form/rated_disabilities` that provides rated rated_disabilities.  We should find a way to allow the veteran to update the rated_disabilities as needed. It seems that `ratedDisabilityId` is the unique id we're interested in comparing to see if there are updates.~

discovery issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/12941
sample data: https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/issues/163 
Back end issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/18681

## updated issue description (copied from comment below):
- Back end will create a new controller specifically for 526 (and probably HLR at some point) 
- Front end refactors in progress logic to call the new controller (requires meeting with platform)
- Back end will pull the new rated disabilities and add to the in progress response (if the rated disabilities are different, the metadata will be updated to change the return URL to direct users to the rated disabilities page)
- Front end adds manipulation logic on the rated disabilities page to handle the changed rated disabilities (theoretically, the places the rated disabilities were previously mentioned throughout the form should be properly updated and handled as the user goes through the form)

**Comment by ndsprinkle** [comment]
This could be tricky because we pull all this information before the form begins in the prefill stage. If this is ultimately possible, then it will be some rewriting of how the form functions and adding the ability to make requests to the api mid form. I'm not sure if this will mean changing the way the form system functions or if there will be a way to just change how our form behaves, but I am going to say this will be some complicated work regardless. 

My only other opinion is that this seems strange how common this is happening because the question on rated disabilities occurs fairly early in the form. I wouldn't expect much to change between starting the form and getting to that question, but, as you point out, the user could take a break from filling out the form before they reach that question and not return for several days. Maybe our best solution will be to just re-pull that portion of the prefill data when the form is continued?

**Comment by ndsprinkle** [comment]
Met with engineering on this and we have landed on a hybrid solution:
- Back end will create a new controller specifically for 526 (and probably HLR at some point)
- Front end refactors in progress logic to call the new controller (requires meeting with platform)
- Back end will pull the new rated disabilities and add to the in progress response (if the rated disabilities are different, the metadata will be updated to change the return URL to direct users to the rated disabilities page)
- Front end adds manipulation logic on the rated disabilities page to handle the changed rated disabilities (theoretically, the places the rated disabilities were previously mentioned throughout the form should be properly updated and handled as the user goes through the form)

**Comment by edmangimelli** [comment]
Going to talk to @cvalarida about approving this FE platform change

**Comment by edmangimelli** [comment]
@cvalarida approves of [proposal](https://github.com/department-of-veterans-affairs/va.gov-team/issues/13128#issuecomment-755604718) 🎉 
https://dsva.slack.com/archives/CQH357ZTP/p1610559643383200?thread_ts=1610558905.381400&cid=CQH357ZTP

**Comment by annaswims** [comment]
I've got https://github.com/department-of-veterans-affairs/vets-api/pull/5717 up.  

tl;dr the `/v0/disability_compensation_in_progress_forms/{id}` endpoints will work the same as the `/v0/in_progress_forms/{id}` endpoints.  The only difference is the get `/v0/disability_compensation_in_progress_forms/{id}` will also update rated disabilities and add a boolean `ratedDisabilitiesUpdated` to the metadata.  If the rated disabilities  have been updated and the veteran is past the rated disability page the returnUrl will be '/disabilities/rated-disabilities'
